lib/pages/notifications/notification_service.dart

```diff
@@ -0,0 +1,89 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
+import 'package:flutter_timezone/flutter_timezone.dart';
+import 'package:timezone/data/latest_all.dart' as tz;
+import 'package:timezone/timezone.dart' as tz;
+import 'package:workmanager/workmanager.dart';
+
+class NotificationsService {
+  static bool _isInitialized = false;
+  static final FlutterLocalNotificationsPlugin
+      _flutterLocalNotificationsPlugin = FlutterLocalNotificationsPlugin();
+
+  static const InitializationSettings _initializationSettings =
+      InitializationSettings(
+    android: AndroidInitializationSettings(
+        '@mipmap/ic_launcher'), // Replace with your actual icon name
+    iOS: DarwinInitializationSettings(),
+  );
+
+  static Future<void> init() async {
+    if (_isInitialized) {
+      // Already initialized, no need to initialize again
+      return;
+    }
+    print("Initialization");
+    tz.initializeTimeZones();
+    // Get the device's current time zone
+    final String timeZoneName = await FlutterTimezone.getLocalTimezone();
+    //  String currentTimeZone = await FlutterNativeTimezone.getLocalTimezone();
+
+    // Set the local time zone
+    tz.setLocalLocation(tz.getLocation(timeZoneName ?? 'Asia/Jerusalem'));
+    await _flutterLocalNotificationsPlugin.initialize(_initializationSettings);
+    _isInitialized = true;
+  }
+
+  static Future<void> showNotification(String title, String body) async {
+    print("trying to show notification");
+    const AndroidNotificationDetails androidNotificationDetails =
+        AndroidNotificationDetails('your channel id', 'your channel name',
+            channelDescription: 'your channel description',
+            importance: Importance.max,
+            priority: Priority.high,
+            ticker: 'ticker');
+    const NotificationDetails notificationDetails =
+        NotificationDetails(android: androidNotificationDetails);
+    await _flutterLocalNotificationsPlugin
+        .show(0, title, body, notificationDetails, payload: 'item x');
+  }
+
+  static Future<void> scheduleNotification(
+      TimeOfDay timeOfDay, String id, String text) async {
+    final tz.TZDateTime now = tz.TZDateTime.now(tz.local);
+    tz.TZDateTime scheduledDate = tz.TZDateTime(tz.local, now.year, now.month,
+        now.day, timeOfDay.hour, timeOfDay.minute);
+
+    if (scheduledDate.isBefore(now)) {
+      scheduledDate = scheduledDate.add(const Duration(days: 1));
+    }
+
+    await _flutterLocalNotificationsPlugin.zonedSchedule(
+        matchDateTimeComponents: DateTimeComponents.time,
```

Comment:
Wouldn't this schedule a static daily notification?

Reply:
Yes, that's the mechanism we are using- repeatedly schedualling a static notification.